### ISSUE TITLE
See if the documentation on how to write a CalcJob plugin is complete

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/aiida-core#3018]
Check if all the content from the [old docs at this link](https://aiida.readthedocs.io/projects/aiida-core/en/stable/developer_guide/devel_tutorial/code_plugin_qe.html) (that used to be a documentation specific to Quantum ESPRESSO) is now in some way explained in the current docs
(in the section [working with CalcJobs](https://aiida-core.readthedocs.io/en/latest/working/calculations.html#calculation-jobs))

**Comment by d-tomerini** [comment]
@giovannipizzi everything is explained in the new section.
There are only two very minor things that I noticed:
- it's not mentioned that ``valid_type`` can be a tuple and not a value
- old documentation warned about the hardcoding of all the variables in the preparation of the input, for maintainability. The ``ArithmeticAddCalculation`` is so simple that this is not mentioned anywhere.

If you consider these important I'll create a PR to address it.

**Comment by giovannipizzi** [comment]
Thanks @d-tomerini! I'm pinging also @sphuber because I think he's been recently working on those sections, and maybe he can answer better to your question

**Comment by sphuber** [comment]
Thanks for going through this @d-tomerini . Regarding the `vaiid_type`, I double checked and in the [calculation job docs](https://aiida-core.readthedocs.io/en/latest/working/calculations.html#define) it indeed only gives example of single type, but directly there I also link to the more [generic and detailed process](https://aiida-core.readthedocs.io/en/latest/working/processes.html#working-processes-spec) documentation that does state that tuples are also accepted.

Regarding the warning about hard-coding: I looked at the original note and don't find it to be very clear. If I understand it correctly it also seems to be a bit specific to the Quantum ESPRESSO example that is used. For me it does not seem necessary to include.

So in short, I think this issue can be closed without PR. @giovannipizzi let me know if that's ok

**Comment by giovannipizzi** [comment]
OK for me, thanks to both!
